crates/bitwarden/src/auth/register.rs

```diff
@@ -0,0 +1,98 @@
+use bitwarden_api_identity::{
+    apis::accounts_api::accounts_register_post,
+    models::{KeysRequestModel, RegisterRequestModel},
+};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::auth_settings::Kdf,
+    crypto::{HashPurpose, MasterKey},
+    error::Result,
+    util::default_pbkdf2_iterations,
+    Client,
+};
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct RegisterRequest {
+    pub email: String,
+    pub name: Option<String>,
+    pub password: String,
+    pub password_hint: Option<String>,
+}
+
+pub(super) async fn register(
+    client: &mut Client,
+    req: &RegisterRequest,
+) -> Result<RegisterResponse> {
```

Comment:
Is it useful to have this response with the master password hash and encryption keys? 

I assume immediately after registering the client is going to log in, and to reuse this information we would need to handle it as a special case, which doesn't seem worth the effort for a one time thing.